### PR TITLE
Add default env for script run

### LINKED ISSUE #4814: Use current deployment context in SCRIPT_RUN stage
**What would you like to be added**:
Use current deployment context (like DeploymentTrigger, DeploymentStatus, etc.) in SCRIPT_RUN stage.
refs: https://pipecd.dev/docs-v0.46.x/user-guide/managing-application/customizing-deployment/script-run/

**Why is this needed**:
Since we don't know about the current deployment, it is difficult to write a script for customization of the deployment process. 

This is similar to GitHubAction, where we can access job and step contexts through a defined property.
refs: https://docs.github.com/en/actions/learn-github-actions/contexts#job-context

### PR DESCRIPTION
**Pull request by ffjlabo**
**What this PR does / why we need it**:

Added default env value for the deployment information (such as deployment id, application id...)

Current values are below.

| Name | Description | Example |
|-|-|-|
|SR_DEPLOYMENT_ID| The deployment id | 877625fc-196a-40f9-b6a9-99decd5494a0 |
|SR_APPLICATION_ID| The application id | 8d7609e0-9ff6-4dc7-a5ac-39660768606a |
|SR_APPLICATION_NAME| The application name | example |
|SR_TRIGGERED_AT| The timestamp when the deployment is triggered  | 1719571113 |
|SR_TRIGGERED_COMMIT_HASH| The commit hash that triggered the deployment | 2bf969a3dad043aaf8ae6419943255e49377da0d |
|SR_REPOSITORY_URL| The repository url configured in the piped config  | git@github.com:org/repo.git, https://github.com/org/repo |
|SR_SUMMARY| The summary of the deployment | Sync with the specified pipeline because piped received a command from user via web console or pipectl|
|SR_CONTEXT_RAW| The json encoded string of above values | {"deploymentID":"877625fc-196a-40f9-b6a9-99decd5494a0","applicationID":"8d7609e0-9ff6-4dc7-a5ac-39660768606a","applicationName":"example","triggeredAt":1719571113,"triggeredCommitHash":"2bf969a3dad043aaf8ae6419943255e49377da0d","repositoryURL":"git@github.com:org/repo.git","labels":{"env":"example","team":"product"}} |
|SR_LABEL_XXX| The label attached to the deployment. The env name depends on the label name. For example, if a deployment has the labels `env:prd` and `team:server`, `SR_LABEL_ENV` and `SR_LABEL_TEAM` are registered.  | prd, server |

PipeCD

example of app.pipecd.yaml
```
apiVersion: pipecd.dev/v1beta1
kind: KubernetesApp
spec:
  name: script-run-like-jenkins
  labels:
    env: example
    team: product
  pipeline:
    stages:
      - name: K8S_CANARY_ROLLOUT
        with:
          replicas: 10%
      - name: SCRIPT_RUN
        with:
          run: |
            echo $SR_DEPLOYMENT_ID
            echo $SR_APPLICATION_ID
            echo $SR_APPLICATION_NAME
            echo $SR_TRIGGERED_AT
            echo $SR_TRIGGERED_COMMIT_HASH
            echo $SR_REPOSITORY_URL
            echo $SR_RAW
            echo $
          env:
            REPOSITORY_URL: "https://github.com/pipecd/pipecd"
          onRollback: |
            echo $SR_DEPLOYMENT_ID
            echo $SR_APPLICATION_ID
            echo $SR_APPLICATION_NAME
            echo $SR_TRIGGERED_AT
            echo $SR_TRIGGERED_COMMIT_HASH
            echo $SR_REPOSITORY_URL
            echo $SR_RAW
...
```

**Which issue(s) this PR fixes**:

Fixes #4814

**Does this PR introduce a user-facing change?**:

- **How are users affected by this change**:
- **Is this breaking change**:
- **How to migrate (if breaking change)**:

### CODE REVIEW

File: pkg/app/piped/executor/scriptrun/scriptrun.go [hunk 1]
```diff
@@ -116,6 +129,55 @@ func (e *Executor) executeCommand() model.StageStatus {
...
+	}
+
+	envs := map[string]string{
+		"SR_DEPLOYMENT_ID":         src.DeploymentID,
+		"SR_APPLICATION_ID":        src.ApplicationID,
+		"SR_APPLICATION_NAME":      src.ApplicationName,
+		"SR_TRIGGERED_AT":          strconv.FormatInt(src.TriggeredAt, 10),
+		"SR_TRIGGERED_COMMIT_HASH": src.TriggeredCommitHash,
+		"SR_REPOSITORY_URL":        src.RepositoryURL,
+		"SR_RAW":                   string(b), // Add the raw json string as an environment variable.
```
Comment (khanhtc1202) [inline pkg/app/piped/executor/scriptrun/scriptrun.go:170]: how about renaming it to `SR_CONTEXT_RAW`

**Review by Warashi — APPROVED** [review]
LGTM

**Comment by ffjlabo** [comment]
oops, I forgot to add the `Summary` value. will try to add it 🙏 
https://github.com/pipe-cd/pipecd/issues/4814#issuecomment-2060583328

**Comment by ffjlabo** [comment]
Added `Summary` value on [701bffa](https://github.com/pipe-cd/pipecd/pull/5002/commits/701bffa8fe819ac82ff594e617a3bd5dc660936c)
PipeCD

**Review by Warashi — APPROVED** [review]
I checked with this PR and [the Deployment model](https://github.com/pipe-cd/pipecd/blob/e20cd0b2ca14b819ce264ecdd6b3ab653cad3307/pkg/model/deployment.proto#L59).

LGTM

**Review by khanhtc1202 — APPROVED** [review]
Here you go :rocket:

**Pull request merged**